File: scratch/fix_export_button.py
import os

file_path = r"c:\Users\benja\OneDrive\Desktop\nutricion_app\frontend\src\app\dashboard\entregable\DeliverableClient.tsx"
with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

# Target strings to replace
target = "disabled={isExporting}"
replacement = "disabled={isExporting || !previousStagesSummary.diet.hasData || !previousStagesSummary.patient.hasData || !previousStagesSummary.recipes.hasData || !previousStagesSummary.cart.hasData}"

if target in content:
    new_content = content.replace(target, replacement)
    with open(file_path, "w", encoding="utf-8") as f:
        f.write(new_content)
    print("Successfully updated the file.")
else:
    print("Target string not found.")

File: scratch/fix_export_button.js
const fs = require('fs');
const path = "c:\\Users\\benja\\OneDrive\\Desktop\\nutricion_app\\frontend\\src\\app\\dashboard\\entregable\\DeliverableClient.tsx";
let content = fs.readFileSync(path, 'utf8');

const target = "disabled={isExporting}";
const replacement = "disabled={isExporting || !previousStagesSummary.diet.hasData || !previousStagesSummary.patient.hasData || !previousStagesSummary.recipes.hasData || !previousStagesSummary.cart.hasData}";

if (content.includes(target)) {
    const newContent = content.split(target).join(replacement);
    fs.writeFileSync(path, newContent, 'utf8');
    console.log("Successfully updated the file components.");
} else {
    console.log("Target string not found in file.");
}
